### PR TITLE
feat: update subscription module with email validation

### PR DESCRIPTION
**Pull request by braianj**

### CODE REVIEW

File: src/modules/subscription/reducer.spec.ts [hunk 1]
```diff
@@ -47,14 +58,52 @@ describe('when reducing the get subscription success action', () => {
...
+        ...state,
+        email: '',
+        subscriptionDetails: objectToCamel(subscription.details),
+        loading: [],
+        error: null
+      })
+    })
+  })
+  describe('and there is an unconfirmedEmail pending', () => {
+    it('should return a state with the default subscription details set without the email and the loading state cleared', () => {
```
Comment (LautaroPetaccio) [inline src/modules/subscription/reducer.spec.ts:73]:
```suggestion
    it('should return a state with the default subscription details set with the unconfirmed email set and the loading state cleared', () => {
```

File: src/modules/subscription/reducer.spec.ts [hunk 2]
```diff
@@ -47,14 +58,52 @@ describe('when reducing the get subscription success action', () => {
       subscriptionDetails: { ...objectToCamel(subscription.details), ignoreAllInApp: true }
     }
   })
+  describe('and there is no email nor unconfirmedEmail', () => {
+    it('should return a state with the default subscription details set without the email and the loading state cleared', () => {
```
Comment (LautaroPetaccio) [inline src/modules/subscription/reducer.spec.ts:62]: Why is it the default subscription details? It's setting the subscription details that came with the action.

File: src/modules/subscription/reducer.spec.ts [hunk 3]
```diff
@@ -47,14 +58,52 @@ describe('when reducing the get subscription success action', () => {
...
 
-  it('should return a state with the  subscription details set, the email set and the loading state cleared', () => {
-    expect(subscriptionReducer(initialState, getSubscriptionsSuccess(subscription))).toEqual({
-      ...state,
-      email: subscription.email,
-      subscriptionDetails: objectToCamel(subscription.details),
-      loading: [],
-      error: null
+  describe('and there is an email but no unconfirmedEmail', () => {
+    it('should return a state with the  subscription details set, the email set and the loading state cleared', () => {
```
Comment (LautaroPetaccio) [inline src/modules/subscription/reducer.spec.ts:86]:
```suggestion
    it('should return a state with the subscription details set, the email set and the loading state cleared', () => {
```

File: src/modules/subscription/reducer.spec.ts [hunk 4]
```diff
@@ -47,14 +58,52 @@ describe('when reducing the get subscription success action', () => {
...
+        email: subscription.email,
+        subscriptionDetails: objectToCamel(subscription.details),
+        loading: [],
+        error: null
+      })
+    })
+  })
+
+  describe('and there is an email and an unconfirmedEmail', () => {
+    it('should return a state with the  subscription details set, the email set and the loading state cleared', () => {
```
Comment (LautaroPetaccio) [inline src/modules/subscription/reducer.spec.ts:98]:
```suggestion
    it('should return a state with the subscription details set, the email and the unconfirmed email set and the loading state cleared', () => {
```

File: src/modules/subscription/reducer.spec.ts (hunk 2 above)
Comment (braianj) [reply src/modules/subscription/reducer.spec.ts]: can you check again?

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**